src/main/java/de/tum/cit/aet/artemis/programming/service/ParticipationVcsAccessTokenService.java

```diff
@@ -73,6 +79,7 @@ public ParticipationVCSAccessToken findByUserAndParticipationIdOrElseThrow(User
     public ParticipationVCSAccessToken createVcsAccessTokenForUserAndParticipationIdOrElseThrow(User user, long participationId) {
         participationVcsAccessTokenRepository.findByUserIdAndParticipationIdAndThrowIfExists(user.getId(), participationId);
         var participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
+        loadTeamStudentsForTeamExercise(participation);
```

Comment:
Can't we load the team/students with the previous call?

Reply:
I think it's fine to load them conditionally, i.e. only when it is a team exercise